engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -658,13 +703,22 @@ static int CreateTexture(lua_State* L)
         max_mipmaps = max_mipmaps_actual;
     }
 
+    dmBuffer::HBuffer buffer = 0;
+    if (dmScript::IsBuffer(L, 3))
+    {
+        // If a buffer has been passed in we can optionally support compression type
+        dmScript::LuaHBuffer* l_buffer = dmScript::CheckBuffer(L, 3);
+        buffer                         = l_buffer->m_Buffer;
+        compression_type               = (dmGraphics::TextureImage::CompressionType) CheckTableInteger(L, 2, "compression_type", (int) dmGraphics::TextureImage::COMPRESSION_TYPE_DEFAULT);
+    }
+
```

Comment:
This is a new option for creating texture that perhaps should be moved out from this PR? 🤔 The reason the option was added is because we don't have a way of creating transcoded data in runtime via the buffer, so to create a compressed texture you first needed to create a dummy texture and then use set_texture to upload the compressed data. That's not optimal and so I think we do want to support passing a buffer to this function that can be used to create a texture with initial data without having to make the dummy texture first..